testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2351,6 +2352,39 @@ public void testRenameSchema()
         }
     }
 
+    @Test
+    public void testDropSchemaCascade()
+    {
+        if (!hasBehavior(SUPPORTS_DROP_SCHEMA_CASCADE)) {
+            String schemaName = getSession().getSchema().orElseThrow();
+            assertQueryFails(
+                    "DROP SCHEMA " + schemaName + " CASCADE",
+                    "This connector does not support dropping schemas with CASCADE option");
+            return;
+        }
+
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_SCHEMA) && hasBehavior(SUPPORTS_CREATE_TABLE));
+
+        String schemaName = "test_drop_schema_cascade_" + randomNameSuffix();
+        String firstTableName = "test_first_table" + randomNameSuffix();
+        String secondTableName = "test_second_table" + randomNameSuffix();
+        try {
+            assertUpdate(createSchemaSql(schemaName));
+            assertUpdate("CREATE TABLE " + schemaName + "." + firstTableName + "(a INT)");
+            assertUpdate("CREATE TABLE " + schemaName + "." + secondTableName + "(a INT)");
```

Comment:
We don't need two tables, one would suffice.

but we should have test cases "drop schema with a view" and  "drop schema with an mview"